src/common/gui/base/Toggle.ts

```diff
@@ -0,0 +1,98 @@
+import m, { ClassComponent, Vnode } from "mithril"
+import { Keys, TabIndex } from "../../api/common/TutanotaConstants.js"
+import { isKeyPressed } from "../../misc/KeyManager.js"
+
+type ToggleVariant = "normal" | "expanded"
+type TogglePosition = "left" | "right"
+type HTMLElementWithAttrs = Partial<Omit<HTMLElement, "style"> & ToggleAttrs>
+
+export interface ToggleAttrs {
+	checked: boolean
+	onclick: (checked: boolean) => unknown
+	ariaLabel: string
+	disabled?: boolean
+	togglePosition?: TogglePosition
+	class?: string
+	variant?: ToggleVariant
+}
+
+/**
+ * Toggle component with variants
+ * @see Component attributes: {ToggleAttrs}
+ * @example
+ * m(Toggle,
+ *     {
+ *         checked: this.checked,
+ *         onclick: (checked: boolean) => {
+ *             this.checked = checked
+ *             console.log(this.checked)
+ *         },
+ *         togglePosition: "right",
+ *         ariaLabel: "Test Toggle",
+ *         disabled: false,
+ *         variant: "normal",
+ *     },
+ *     "My label",
+ * ),
+ */
+
+export class Toggle implements ClassComponent<ToggleAttrs> {
+	private checkboxDom?: HTMLInputElement
+
+	view({ attrs: { disabled, variant, ariaLabel, checked, onclick, togglePosition }, children }: Vnode<ToggleAttrs>) {
+		const childrenArr = [children, this.buildTogglePillComponent(onclick, disabled)]
+		if (togglePosition === "left") {
+			childrenArr.reverse()
+		}
+
+		return m(
+			"label.toggle",
+			{
+				class: this.resolveClasses(disabled, variant),
+				role: "checkbox",
```

Comment:
> Toggle button: A two-state button that can be either off (not pressed) or on (pressed). To tell assistive technologies that a button is a toggle button, specify a value for the attribute [aria-pressed](https://w3c.github.io/aria/#aria-pressed). For example, a button labelled mute in an audio player could indicate that sound is muted by setting the pressed state true. Important: it is critical the label on a toggle does not change when its state changes. In this example, when the pressed state is true, the label remains "Mute" so a screen reader would say something like "Mute toggle button pressed". Alternatively, if the design were to call for the button label to change from "Mute" to "Unmute," the aria-pressed attribute would not be needed.

- https://www.w3.org/WAI/ARIA/apg/patterns/button/

Should we do this as an Aria Button instead?

Reply:
Actually, we should use this one https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/